### PR TITLE
workflows/staging-tests: open an issue on failure

### LINKED ISSUE #161: Notify on CI failure
**Description**

We have some tests that run on a schedule to ensure the client continues to work against Sigstore's staging instance, but we have no way to be notified if they start failing, and we'll likely move these to only run on main/cron and not on PRs (ref: https://github.com/sigstore/sigstore-python/pull/159).

We should have some way to notify maintainers when these jobs start failing, perhaps by automatically opening an issue on this repository, or email notifications.

### PR DESCRIPTION
**Pull request by woodruffw**
Closes #161.

Signed-off-by: William Woodruff <william@trailofbits.com>

### CODE REVIEW

File: .github/workflows/staging-tests.yml [hunk 1]
```diff
@@ -37,3 +42,32 @@ jobs:
...
+
+      - name: open an issue if the staging tests fail
+        if: failure()
+        uses: peter-evans/create-issue-from-file@v4
+        with:
+          title: "[CI] Integration failure: staging instance"
+          # created in the previous step
+          content-filepath: /tmp/staging-instance-issue.md
+          labels: bug,component:cicd,component:tests
+          assignees: woodruffw,di,tetsuo-cpp
```
Comment (woodruffw) [inline .github/workflows/staging-tests.yml:73]: Flagging: I put the three of us as assignees, but maybe someone from Fulcio or Rekor also makes sense?

**Comment by woodruffw** [comment]
One thing: this will create a new issue each time the staging tests fail, which could happen several times (on the `cron` schedule) before a fix is implemented and applied. I'm not sure if that's worth addressing now.

File: .github/workflows/staging-tests.yml (hunk 1 above)
Comment (di) [reply .github/workflows/staging-tests.yml:73]: I think this is fine for us to triage.

**Comment by di** [comment]
> One thing: this will create a new issue each time the staging tests fail, which could happen several times (on the `cron` schedule) before a fix is implemented and applied. I'm not sure if that's worth addressing now.

Might get annoying if staging is broken for a long time, but I figure we can cross that bridge when we get to it.

File: .github/workflows/staging-tests.yml [hunk 2]
```diff
@@ -37,3 +42,32 @@ jobs:
...
+
+          This suggests one of three conditions:
+
+          * A backwards-incompatible change in a Sigstore component;
+          * A regression in \`sigstore python\`;
+          * A transient error.
+
+          The full CI failure can be found here:
+
+          $GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID
```
Comment (di) [inline .github/workflows/staging-tests.yml:62]: Nice!

**Review by di — APPROVED** [review]

**Comment by jku** [comment]
> > One thing: this will create a new issue each time the staging tests fail, which could happen several times (on the `cron` schedule) before a fix is implemented and applied. I'm not sure if that's worth addressing now.
> 
> Might get annoying if staging is broken for a long time, but I figure we can cross that bridge when we get to it.

when you do, a rough bridge is here: https://github.com/theupdateframework/python-tuf/blob/develop/.github/workflows/specification-version.yml :wink:

**Comment by woodruffw** [comment]
Nice! Yeah, avoiding a third party action where we can do so conveniently seems like an easy win to me.

**Pull request merged**